test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -3,240 +3,264 @@
             {
-                await connection.StartAsync();
+                var httpConnection = new HttpConnection(new Uri(_serverFixture.BaseUrl + "/hubs"), transport, loggerFactory);
+                var connection = new HubConnection(httpConnection, protocol, loggerFactory);
+                try
+                {
+                    await connection.StartAsync().OrTimeout();
 
-                var result = await connection.InvokeAsync<string>(nameof(TestHub.HelloWorld));
+                    var result = await connection.InvokeAsync<string>(nameof(TestHub.HelloWorld)).OrTimeout();
 
-                Assert.Equal("Hello World!", result);
-            }
-            finally
-            {
-                await connection.DisposeAsync();
+                    Assert.Equal("Hello World!", result);
+                }
+                catch (Exception ex)
+                {
+                    loggerFactory.CreateLogger<HubConnectionTests>().LogError(ex, "Exception from test");
+                    throw;
+                }
+                finally
+                {
+                    await connection.DisposeAsync().OrTimeout();
+                }
             }
         }
 
         [Theory]
-        [MemberData(nameof(HubProtocols))]
-        public async Task CanSendAndReceiveMessage(IHubProtocol protocol)
+        [MemberData(nameof(HubProtocolsXTransports))]
+        public async Task CanSendAndReceiveMessage(IHubProtocol protocol, TransportType transport)
         {
-            var loggerFactory = CreateLogger();
-            const string originalMessage = "SignalR";
-
-            var httpConnection = new HttpConnection(new Uri("http://test/hubs"), TransportType.LongPolling, loggerFactory, _testServer.CreateHandler());
-            var connection = new HubConnection(httpConnection, protocol, loggerFactory);
-            try
+            using (StartLog(out var loggerFactory))
             {
-                await connection.StartAsync();
+                const string originalMessage = "SignalR";
 
-                var result = await connection.InvokeAsync<string>(nameof(TestHub.Echo), originalMessage);
+                var httpConnection = new HttpConnection(new Uri(_serverFixture.BaseUrl + "/hubs"), transport, loggerFactory);
+                var connection = new HubConnection(httpConnection, protocol, loggerFactory);
+                try
+                {
+                    await connection.StartAsync().OrTimeout();
 
-                Assert.Equal(originalMessage, result);
-            }
-            finally
-            {
-                await connection.DisposeAsync();
+                    var result = await connection.InvokeAsync<string>(nameof(TestHub.Echo), originalMessage).OrTimeout();
+
+                    Assert.Equal(originalMessage, result);
+                }
+                catch (Exception ex)
+                {
+                    loggerFactory.CreateLogger("Test").LogError(ex, "Exception from test");
```

Comment:
These were the changes I lost... I remember fixing this and thought I missed one of the occurrences. Yesterday when changing branches I noticed HubConnectionTests.cs was not saved and VS unloaded it. I was wondering what I lost and now I know.